src/editors/_shared/hooks/DataTypes/testUtils.js

```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import { withTheme } from "@mov-ai/mov-fe-lib-react";
+import { render, screen, fireEvent } from "@testing-library/react";
+
+export function getRendered(type, props = {}) {
+  const Component = type.getEditComponent();
+  const Themed = withTheme(Component);
+  const { container } = render(<Themed { ...props } />);
+  expect(container).toBeInTheDocument();
+}
+
+export function testStringInput(DataType, stringValue, realValue) {
+  it("Renders correctly", () => {
+    getRendered(new DataType());
+  });
+
+  it("Set value correctly", () => {
+    const onChange = jest.fn();
+    getRendered(
+      new DataType(), {
+        rowData: { value: [] },
+        onChange,
+      }
+    );
+    const input = screen.getByTestId('input_value');
+    fireEvent.change(input, { target: { value: stringValue } });
```

Comment:
`fireEvent` is not recommended. Please use `userEvent`. As mentioned [here](https://blog.mimacom.com/react-testing-library-fireevent-vs-userevent/)
TL;DR; 
**So, what's the difference between them? Well, the difference is that userEvent simulates complete interactions, and not only single events.**

Reply:
I don't want a complete interaction as I want the mock function to be called only once. Not once for every letter.

Reply:
I still think we need to test with userEvent instead of fireEvent. The more close to reality we get, the more resilient the test.